runtime/oti/j9protos.h

```diff
@@ -1133,6 +1133,52 @@ extern J9_CFUNC struct J9Method* jitResolveStaticMethodRef(J9VMThread *vmStruct,
 extern J9_CFUNC struct J9Method* jitResolveSpecialMethodRef(J9VMThread *vmStruct, J9ConstantPool *constantPool, UDATA cpOrSplitIndex, UDATA resolveFlags);
 extern J9_CFUNC struct J9Class * jitGetDeclaringClassOfROMField(J9VMThread *vmStruct, J9Class *clazz, J9ROMFieldShape *romField);
 
+typedef struct J9MethodFromSignatureWalkState {
+	U_8 *className;
```

Comment:
I suggest the type of this field should be `const char *` for consistency with name fields in `J9JNINameAndSignature`.

Reply:
So I tried to play around with this and it seems we cannot make this change or any of the below changes to types without casting away `const` because making `classsName` `const char *` means that we cannot feed this value into `internalFindClassUTF8` because that API expects a `U_8 *` so we have to cast away constness there.

We can however do it with `methodName` and `methodSig` parameters. So the question becomes, do we want to have an asymmetrical API in which the class name is not `const` qualified, but method name and signature are?

In summary we can do:
- Keep as it is today
- Change `methodName` and `methodSig` to `const U_8 *` or simply `const char *`, but leave `className` without `const` qualifier
- Add constness to all three name arguments and cast away to stay consistent but cast away constness when using `internalFindClassUTF8`

Please advise which we should prefer.

Reply:
I think we should make everything `const` that can be. Adding the (hopefully) occasional cast is preferable (IMO) to expanding the mass of code that declares more access than it needs.